### ISSUE TITLE
[CoE Starter Kit - QUESTION] Exclude Environments and or Components

### ISSUE THREAD

**Issue by GarryPope** [microsoft/coe-starter-kit#8169]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

Hello CoE Team,

I hope you are well. 

Is there a way to exclude environments or individual components, such as apps or flows etc, from the Power BI report via the Power Platform Admin View app, please? 

How I would see it working as, there is a column on the record such as "Exclude from Power BI Report" where I toggle to "Yes".

- Component: If I toggle it to "Yes" on a component record, say a Power App, then when I view the Power BI Report the app doesn't not show on the report. 
- Environment: And if I was to do it at an environment level, all components in the environment don't show in the Power BI Report.

The benefits are that the Power BI report gives a clearer picture of the data. An example would be we'd like to exclude the "Dataverse Accelerator App" from the Power BI Report. This appears in what looks like every environment. When I report the stats of how many components have been built, if we created 5 environments then this shows 5 times, but really the business is getting no "value" from this app. 

The other benefit is if I have a Dev, Test, QA, Preprod, Prod environment, and I build a Flow that saves 1 FTE in production, my statistics show that I have 5 Flows, where really, the business value is in the 1 Flow in production, so I'd like to exclude the Dev, Test, UAT, Preprod environments. 

In a nutshell I want to report "real" components that provide value in order to track the real success of the Power Platform. 

Hope that makes sense? Sorry to write so much!

Garry

### What solution are you experiencing the issue with?

None

### What solution version are you using?

February 2024

### What method are you using to get inventory and telemetry?

None

AB#3072

**Comment by Jenefer-Monroe** [comment]
Hello, no this does not exist today. Its a good suggetion though. Will put on the backlog for near future consideration.

**Comment by GarryPope** [comment]
Hello CoE Team,

I was reading through your documentation and saw this "[Setting up CoE for a subset of environments"](https://learn.microsoft.com/en-us/power-platform/guidance/coe/faq#setting-up-coe-for-a-subset-of-environments). 

I'm hoping this will exclude environments such as Dynamics Dev, Test, UAT, Staging, Preprod etc. This will hopefully solve one of my requirements. 

Will give it a try. If it isn't successful, I'll message back here. 

Thanks very much!

Garry

**Comment by Jenefer-Monroe** [comment]
Sorry I thought you had already knew bout those and they weren't quite what you were looking for. Thanks for continuing to explore!

**Comment by GarryPope** [comment]
Hello @Jenefer-Monroe,

Thanks for the reply, you're quite right, that is not exactly what I'm looking for, but I thought this might be a good workaround to at least exclude entire environments. 

It works really well on initial set up of the CoE, so I can exclude entire environments, but for any new environments created, unless I change the value on the environment record in the column "Excuse From Inventory" to "Yes" I get all the component records in the CoE Admin app, and of course I then see them in the Power BI report. I could create a Flow so as soon an an environment is created I toggle the value to "Yes", and get alerted on record creation as to whether I toggle it back to "No", but I try to not customise the CoE at all, and use the out of the box functionality

Either way, I still think my idea of not showing an environment and its components or just an individual component in the Power BI report is the best option.

Like always, thanks for your excellent support.

Garry

**Comment by Jenefer-Monroe** [comment]
Actually that behavior is just bug, when we switched from v3 to v4 of the Driver I forgot to update that field. 
Please see https://github.com/microsoft/coe-starter-kit/issues/7953 where this is fixed in the April release. 
So on your next upgrade that will behave as you wish. 
With that fixed, would your need be resolved?

**Comment by Jenefer-Monroe** [comment]
Looks like this got left open accidently and is fixed via the issues shown.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
